### PR TITLE
chore: remove legacy app-engine code

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

The way we currently handle taking backups on the PP site is through a weekly task defined in functions\src\exports\tasks.ts and triggered using google cloud pubsub. The pubsub integration was manually setup right at the beginning of the project, and would not be automatically managed for new deployments

The only reason we did things in this way was because cloud functions previously never had support for defining cron triggers, so we had created the functions-cron folder which created a google app-engine instance (which does have cron) to spin-up and publish a cloud pub-sub message on a weekly and daily basis which the functions could be triggered by. In order to trigger on a different project we would need to deploy new app engine instances and add the pubsub events manually through the cloud console page.

Firebase finally added support for scheduled functions a bit over a year ago so actually all this system can finally be thrown out and replaced by much simpler functions (that would also get copied over to any other instance we deploy to). 

Changes:

- Remove legacy functions-cron code and readme notes
- Refactor existing pubsub daily and weekly functions to use scheduled functions
- Rename folder containing scheduled functions to be called `scheduled` (previously `exports`)

## Review Notes
The only real way to test is to deploy to staging and check at least the daily tick is triggered successfully (scheduled for 2am UTC daily). 

There is a chance deployment will throw an error if specific apis need to be enabled on google cloud, e.g. [pubsub API](https://console.cloud.google.com/apis/library/pubsub.googleapis.com), or if the existing firebase service account needs additional permissions (there have been quite a few changes over the years so hard to know for sure what is/isn't included by default or within our current setups) 

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on slack in the `platform-dev` channel.

### CODE REVIEW

File: functions/src/scheduled/tasks.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Scripts executed on a cron schedule, such as daily or weekly
+https://firebase.google.com/docs/functions/schedule-functions
+************************************************************************************/
+
+import * as functions from 'firebase-functions'
+import { BackupDatabase } from '../Firebase/databaseBackup'
+import * as FirebaseSync from '../Firebase/firebaseSync'
+
+/** Trigger tasks weekly on Sunday at 2am */
+export const weeklyTasks = functions.pubsub
```
Comment (chrismclarke) [inline functions/src/scheduled/tasks.ts:11]: This code had been mostly just copied from the removed `src/exports/tasks.ts` file, git just doesn't track the changes fully

File: functions/src/scheduled/tasks.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  .schedule('0 2 * * SUN')
+  .onRun(async (context) => {
+    console.log('weekly tick', context)
+    const backupStatus = await BackupDatabase()
+    console.log(backupStatus)
```
Comment (thisislawatts) [inline functions/src/scheduled/tasks.ts:16]: **thought:** I think there is probably value in this being emitted to a Slack channel as a signal of a healthy system. Not something to be done as part of this PR but a useful enhancement going forward.

File: .eslintignore [hunk 3]
```diff
@@ -1,6 +1,5 @@
 node_modules
 functions
-functions-cron
```
Comment (thisislawatts) [inline .eslintignore:3]: **praise:** It is really good to clean this directory out of the root. Very happy to see this change!

**Review by thisislawatts — APPROVED** [review]

File: functions/src/scheduled/tasks.ts (hunk 2 above)
Comment (chrismclarke) [reply functions/src/scheduled/tasks.ts:16]: agreed, would be nice if there were some sort of integration that would at least pass any errors flagged from the functions to some channel somewhere (pretty sure these logs have no value in any terms, but I'll leave for now to at least check output for first run)

**Pull request merged**